service/controller/v15/resource/encryptionkey/create.go

```diff
@@ -8,45 +8,34 @@ import (
 	"github.com/giantswarm/aws-operator/service/controller/v15/key"
 )
 
-func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	customObject, err := key.ToCustomObject(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	createInput, err := toEncryptionKeyState(createChange)
-	if err != nil {
-		return microerror.Mask(err)
-	}
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring encryption key")
 
-	if createInput.KeyName != "" {
-		err := r.encrypter.CreateKey(ctx, customObject, createInput.KeyName)
+		current, err := r.encrypter.CurrentState(ctx, customObject)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", "creating encryption key: created")
-	} else {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "creating encryption key: already created")
-	}
-
-	return nil
-}
-
-func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
-	currentKeyState, err := toEncryptionKeyState(currentState)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
+		desired, err := r.encrypter.DesiredState(ctx, customObject)
+		if err != nil {
+			return microerror.Mask(err)
+		}
 
-	desiredKeyState, err := toEncryptionKeyState(desiredState)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
+		if current.KeyName == "" {
+			err = r.encrypter.CreateKey(ctx, customObject, desired.KeyName)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
 
-	if currentKeyState.KeyName == "" || desiredKeyState.KeyName != currentKeyState.KeyName {
-		return desiredKeyState, nil
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensured encryption key")
```

Comment:
Do we go now with the "ensure" message for logs here or is it the "creating" messaging? I feel like there are some differences in what we express and which information we provide. The latter option would also give information on if the creation was done in this very reconciliation loop or not. With the former we only know it is a done deal but we do not know when and how and kind of it at all.

Reply:
I tried to add something here https://github.com/giantswarm/aws-operator/pull/1096/commits/68823c6342a5bc65355021bcadfedb8d0d9448be

The problem is this resource doesn't gather facts. I will try to improve that further, as this should what `addRoleAccess` does currently (see description).